src/compiler/performance.ts

```diff
@@ -1,146 +1,209 @@
 /*@internal*/
 /** Performance measurements for the compiler. */
-namespace ts.performance {
-    let perfHooks: PerformanceHooks | undefined;
-    // when set, indicates the implementation of `Performance` to use for user timing.
-    // when unset, indicates user timing is unavailable or disabled.
-    let performanceImpl: Performance | undefined;
-
-    export interface Timer {
+namespace ts {
+    interface Timer {
         enter(): void;
         exit(): void;
     }
 
-    export function createTimerIf(condition: boolean, measureName: string, startMarkName: string, endMarkName: string) {
-        return condition ? createTimer(measureName, startMarkName, endMarkName) : nullTimer;
+    export interface Statistic {
+        name: string;
+        value: number;
+        type: StatisticType
+    }
+
+    export enum StatisticType {
+        time,
+        count,
+        memory,
     }
 
-    export function createTimer(measureName: string, startMarkName: string, endMarkName: string): Timer {
-        let enterCount = 0;
+    const nullTimer: Timer = { enter: noop, exit: noop };
+    export const performance = createPerformanceTracker();
+    export const buildPerformance = createPerformanceTracker();

```

Comment:
Is there a reason we need two different performance instances? Do these two overlap WRT mark and measure names?  

The original `ts.performance` API was designed to mirror the native User Timings API and essentially act as an adapter between the DOM and NodeJS implementations, but the user timings API doesn't support multiple instances. The benefit of mirroring the published API (and using it whenever possible) is that we can leverage existing performance analysis tools.